WordPress/src/main/java/org/wordpress/android/ui/mysite/QuickStartRepository.kt

```diff
@@ -72,6 +75,9 @@ class QuickStartRepository
     val activeTask = _activeTask as LiveData<QuickStartTask?>
 
     private var pendingTask: QuickStartTask? = null
+    private var pendingCategoryCompletion: QuickStartTaskType? = null
```

Comment:
The solution with the pending category seems a little bit messy :-). I guess the reason why we can't show the snackbar right when the task is completed is that we can't differentiate what screen we're on, right? Normally I'd expect the livedata to be only consumed when the fragment is visible but I guess that doesn't work in our case.

Reply:
>I guess the reason why we can't show the snackbar right when the task is completed is that we can't differentiate what screen we're on, right?

Yes. But even if we're actually on the My Site screen at the time of completion, that would still be an issue considering some tasks are completed _right before we leave_ the screen. The [`EXPLORE_PLANS` task](https://github.com/wordpress-mobile/WordPress-Android/blob/bde6762d259976921c7dbae6e48d88fe8e1ed8f3/WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt#L269-L272) is one example.

>Normally I'd expect the livedata to be only consumed when the fragment is visible but I guess that doesn't work in our case.

That's correct. The snackbar livedata can be consumed even if the fragment is not visible. It only stops being observed once the fragment hits `onStop`.

I guess we could try to find a more elegant solution after we deal with #14073 as well. Feel free to share any suggestions you may have.